test/tests/functional/pbs_resource_usage_log.py

```diff
@@ -0,0 +1,350 @@
+        self.server.rerunjob(jobid=jid)
+        self.server.rerunjob(jobid=jid_a)
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid1)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid2)
+
+        self.server.delete(jid, wait=True)
+        self.server.delete(jid_a, wait=True)
+
+        # Check for R records for every qrerun
+        self.server.accounting_match(
+            msg='.*R;' + jid +
+            '.*Exit_status=-11.*resources_used.*run_count=1', id=jid,
+            regexp=True)
+        self.server.accounting_match(
+            msg='.*R;' + jid +
+            '.*Exit_status=-11.*resources_used.*run_count=2', id=jid,
+            regexp=True)
+        self.server.accounting_match(
+            msg='.*E;' + jid +
+            '.*Exit_status=271.*resources_used.*run_count=3', id=jid,
+            regexp=True)
+
+        matches = self.server.accounting_match(
+            msg='.*R;' + re.escape(subjid1) +
+            '.*Exit_status=-11.*resources_used.*run_count=1',
+            id=subjid1, regexp=True, allmatch=True)
+        self.assertEqual(len(matches), 2, "Expected 2 R records")
+        matches = self.server.accounting_match(
+            msg='.*R;' + re.escape(subjid1) +
+            '.*Exit_status=-11.*resources_used.*run_count=1',
+            id=subjid2, regexp=True, allmatch=True)
+        self.assertEqual(len(matches), 2, "Expected 2 R records")
+        self.server.accounting_match(
+            msg='.*E;' + re.escape(jid_a) +
+            '.*Exit_status=1.*run_count=0',
+            id=jid_a, regexp=True)
+
+    def test_acclog_force_requeue(self):
+        """
+        Check for resource usage when job is force requeued
+        """
+        a = {'Resource_List.select': '1:ncpus=1:mem=200gb'}
+        j1 = Job(TEST_USER, a)
+        j1.create_eatcpu_job()
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid1)
+
+        # kill -9 mom
+        self.mom.signal('-KILL')
+
+        # Verify that nodes are reported to be down.
+        self.server.expect(NODE, {ATTR_NODE_state: 'down'},
+                           id=self.mom.shortname)
+        self.server.rerunjob(jid1, extend='force')
+
+        # Look for R record as job was force requeued
+        self.server.accounting_match(
+            msg='.*R;' + jid1 + '.*Exit_status=0.*resources_used.*run_count=1',
+            id=jid1, regexp=True)
+
+    def test_acclog_services_restart(self):
+        """
+        Check for resource usage in accounting logs after
+        PBS services are restarted
+        """
+        a = {'Resource_List.select': '1:ncpus=1:mem=200gb'}
+        j1 = Job(TEST_USER, a)
+        j1.create_eatcpu_job(60)
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid1)
+
+        # Restart PBS services
+        PBSInitServices().restart()
+        time.sleep(40)
```

Comment:
Please add a comment stating why the test needs to sleep for 40.

Reply:
I'd rather see a self.logger.info() instead of a comment.  It serves the same purpose and helps the test runner know why their test execution has stalled.

Reply:
Done.